third_party/move/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -153,6 +156,18 @@ impl OnDiskCompiledPackage {
                 deps_compiled_units.push((dep_name, self.decode_unit(dep_name, &bytecode_path)?))
             }
         }
+        let mut bytecode_deps = vec![];
```

Comment:
I'm a bit confused here -- isn't into_compiled_package only used by collect_coverage in the Move CLI? But the bug seems to be about deployment.

Reply:
This part is irrelevant to this fix. Removed.

Since testing this requires publishing a package with bytecode dependencies which is hard to be done automatically, we tested this maually.